drtprod: fetch the DD API KEY

nameisbhaskar opened the pull request:
This PR fetches and sets the DD_API_KEY env if it is possible. This is needed for audit logging all events by drtprod.

Epic: None
Release: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/141895)
<!-- Reviewable:end -->

nameisbhaskar committed:
drtprod: fetch the DD API KEY

This PR fetches and sets the DD_API_KEY env if it is possible. This is needed for audit logging all events by drtprod.

Epic: None
Release: None

nameisbhaskar commented:
Thanks @shailendra-patel nd @vidit-bhat for the review

nameisbhaskar commented:
bors r=@shailendra-patel,@vidit-bhat
